edb/edgeql/compiler/viewgen.py

```diff
@@ -922,11 +922,21 @@ def _normalize_view_ptr_expr(
             specified_required = shape_el.required
         else:
             specified_cardinality = base_cardinality
-            specified_required = base_required
 
-            if (shape_el.cardinality is not None
-                    and base_ptrcls is not None
-                    and shape_el.cardinality != base_cardinality):
+            # Inferred optionality overrides that of the base pointer
+            # if pointer is not `required`.
+            if shape_el.required is not None:
+                specified_required = shape_el.required
+            elif base_required is True:
```

Comment:
Why only if it's true?

Reply:
Because we still want `foo := 1` to infer as `ONE` even if `foo` is defined as `optional` in the base type.  I'll clarify the comment above.